Added border for tabs + container.

toger5 opened the pull request:
the highlighting also works for them main viewport now. since they are using the same sb than the other tabs.

<img width="221" alt="screen shot 2017-09-02 at 19 37 16" src="https://user-images.githubusercontent.com/16718859/29997541-3e923f4c-9016-11e7-86ef-0b8297801e61.png">
<img width="516" alt="screen shot 2017-09-02 at 19 37 28" src="https://user-images.githubusercontent.com/16718859/29997540-3e90f178-9016-11e7-9911-c55fdc5e0bec.png">

 - moved tabConetent position to be more reasonable.
 - fixed drawing order to allow those kind of styles.

## [DISCUSSION] all figured out so ready to merge

this pr is a discussion to some extend. I wanted to propose the border. but I don't know if it makes sense. It would be really easy to make it another option too. (the border already is dependent on the setting you choose for border width. setting it to 0 removes it of course)
 - The drawing order fix though makes much more sense since it allows to draw tabs over the container so more style options are possible in general. (also for games made with godot)
 - also the change, that the same sb is used for the viewport tabs and the dock tabs is good it think.

this is how it looks with the light theme:
<img width="1680" alt="screen shot 2017-09-02 at 19 42 28" src="https://user-images.githubusercontent.com/16718859/29997569-e9afc502-9016-11e7-995e-2de44056bfbc.png">

toger5 committed:
Added border for tabs + container.
 - moved tabConetent position to be more reasonable.
 - fixed drawing order to allow those kind of styles.

djrm commented:
@toger5 i like it, however, i would use the same color as the trees border, which is more subtle

toger5 commented:
@djrm with the light theme that might work really well. I dont know about a dark theme which uses a really dark xolor. It could look amazing. I will definitly try some more variations

djrm commented:
@toger5 it is going to look fine on dark theme, it will be just like a shadow

toger5 commented:
@djrm it is SUPER subtle
<img width="253" alt="screen shot 2017-09-04 at 18 26 34" src="https://user-images.githubusercontent.com/16718859/30033938-a0733b10-919e-11e7-813d-5d9b409cd152.png">

I think we should stick with the contrast color.

toger5 commented:
Another idea which would fit well into that change is, using the highlight color for highlight tabs.
(I'm using the HIGHLIGHT_COLOR_BG version which is a little lighter/darker dependent on the theme -> also less saturated)

would look like this:
<img width="282" alt="screen shot 2017-09-04 at 18 37 28" src="https://user-images.githubusercontent.com/16718859/30034284-a783233c-91a0-11e7-8a27-b53d2060b017.png">

compared to:
<img width="260" alt="screen shot 2017-09-04 at 18 46 05" src="https://user-images.githubusercontent.com/16718859/30034392-531c0e98-91a1-11e7-9ef2-c592bc6039f3.png">

toger5 commented:
for now I just rebased and changed nothing until we decided on.
 - [ ] highlighcolor for tab highlighting (the theme highlight color)
 - [ ] dark theme = dark border OR dark_theme = light border (for light theme the opposite)

my opinions are: 
<img width="242" alt="screen shot 2017-09-04 at 18 51 54" src="https://user-images.githubusercontent.com/16718859/30034493-255f5cf2-91a2-11e7-8f2a-51f426962d35.png">

contrast color for border + no highlight color for highlight tabs (highlight tabs just become interpolated with font color + base_color)

djrm commented:
@toger5 so whats the end result of this PR, can you post the current status on the main PR.

toger5 commented:
@djrm it is the last screenshot I posted

djrm commented:
@toger5 the last for dark, and the last for white?

toger5 commented:
@djrm oh you are right I also should update a screenshot for the dark version after this pr

toger5 commented:
Here are the final results after this pr:
<img width="242" alt="screen shot 2017-09-04 at 18 51 54" src="https://user-images.githubusercontent.com/16718859/30034493-255f5cf2-91a2-11e7-8f2a-51f426962d35.png">

<img width="279" alt="screen shot 2017-09-04 at 19 41 30" src="https://user-images.githubusercontent.com/16718859/30035313-218790ca-91a9-11e7-9523-307735443c6a.png">
<img width="279" alt="screen shot 2017-09-04 at 19 41 41" src="https://user-images.githubusercontent.com/16718859/30035315-21a92f00-91a9-11e7-8266-9ec4ae8d0187.png">
<img width="273" alt="screen shot 2017-09-04 at 19 41 52" src="https://user-images.githubusercontent.com/16718859/30035314-21a740e6-91a9-11e7-9d10-16c2e5bc83e0.png">

(but the lines will get a little bit less saturated)

djrm commented:
@toger5 i dont like, the way it looks on dark theme, and also you will get a lot of complaints, i tried that once (thats the history behind highlighted tabs) and i saw general rejection (and at some extend i can see why), it looks kinda intrusive. i would go for the option i proposed, it is very subtle but it helps to differentiate the areas, and thats what we need, just a subtle separator between areas

toger5 commented:
@djrm ppl which want subtle separators will just turn it of entirely I guess.
I like subtle separators and I use border width 0. since the color difference is enough of a separator for my taste.
what about making it an option?  (maybe that is too many options... I don't know)
but making it dark would defeat it's preps since it would be almost invisible.
But leaving it could result in ppl not liking it. and than disabling border. which would make other parts worse...

I think since the light theme fixes are merged now, I will upload screenshots how it looks with a little darker borders.

Than I give you (@djrm) the power to decide. When no-one else joins the discussion. I can't make a reasonable argument for the lighter border except that I like it more. (which is not a convincing point...)

thoughts on highlight tabs using highlight color?

djrm commented:
@toger5 the default border size is 1, so disabling it would require to use border 0 which im still doubting about, but anyways, the whiter color is more intrusive, i already dealt with comments about it, so it would be better to just use something more subtle (like the color i proposed), also if people want a more saturated border color for panels they can turn on hightlight tabs. about the colored tabs, i dont see the reason for them (isnt simply highlighting them enough) having too much colors and setting is not very good, in my opinion.

toger5 commented:
marked the second discussion point as solved. so will not make the color for highlight tab change

toger5 commented:
current version which is a little bit more saddle
<img width="266" alt="screen shot 2017-09-07 at 00 08 44" src="https://user-images.githubusercontent.com/16718859/30136767-c677ef68-9360-11e7-9423-0f0853558a96.png">
previous version:
<img width="516" alt="screen shot 2017-09-02 at 19 37 28" src="https://user-images.githubusercontent.com/16718859/29997540-3e90f178-9016-11e7-9911-c55fdc5e0bec.png">

toger5 commented:
@djrm now you can decide. But is guess we still go back to the really dark color.

djrm commented:
@toger5, I still think the dark border is a better idea.

this was my proposal some time ago, the borders are even more subtle than in your implementation, and people complained (now i understand why).
![screenshot from 2017-07-14 23-45-32](https://user-images.githubusercontent.com/1103897/30176528-5e7ca51c-93c8-11e7-9e3e-27027f234090.png)

toger5 commented:
@djrm cool so i wull change to dark_color 3

toger5 commented:
just updated with dark_color_3

djrm commented:
@toger5 can you post the latest status of this, and possibly fix conflicts

toger5 commented:
@djrm 
<img width="539" alt="screen shot 2017-09-15 at 00 59 21" src="https://user-images.githubusercontent.com/16718859/30459468-28bc599c-99b1-11e7-9141-fe7008de1389.png">
<img width="537" alt="screen shot 2017-09-15 at 00 59 10" src="https://user-images.githubusercontent.com/16718859/30459469-28df4d94-99b1-11e7-8529-4379c774c067.png">

djrm reviewed (approved):
nice
